### PR TITLE
Qna support in vscode extension

### PR DESCRIPTION
**Pull request by feich-ms**
Add vscode extension support for qna files, including highlight, diagnostic and auto-completion.

# Auto-completion
qna_completion

# Highlight
qna_syntax_highlighting

# Diagnostics
qna_diagnostic

### CODE REVIEW

**Comment by chrimc62** [comment]
>     if(fileName === undefined || !fileName.toLowerCase().endsWith('.lu')) {


Minor comment--our style has blank lines in between functions.

---
Refers to: experimental/adaptive-tool/client/src/extension.ts:252 in 62152da. [](commit_id = 62152da1bdc1da3b829b51c74c8bce15cec35a08, deletion_comment = False)

File: experimental/adaptive-tool/server/qna/src/providers/diagnostics.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import * as path from 'path';
+
+import { TextDocument } from 'vscode-languageserver-textdocument';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const parseFile = require('@microsoft/bf-lu/lib/parser/lufile/luParser');
+
+
+
+export async function updateDiagnostics(document: TextDocument, connection: Connection): Promise<void> {
```
Comment (chrimc62) [inline experimental/adaptive-tool/server/qna/src/providers/diagnostics.ts:20]: Why all the blank lines?

**Review by chrimc62 — APPROVED** [review]
:shipit:

File: experimental/adaptive-tool/server/qna/src/providers/diagnostics.ts (hunk 1 above)
Comment (feich-ms) [reply experimental/adaptive-tool/server/qna/src/providers/diagnostics.ts:16]: Thanks for the review. Optimized the style issues.

**Pull request merged**